### PR TITLE
Fix secrets creation rule for standard user

### PR DESCRIPTION
**Pull request by nickgerace**
Add secrets creation rule for restrictedAdmin. This applies to all
apiGroups.

Issue: #29379

### CODE REVIEW

**Comment by nickgerace** [comment]
Admittedly, I am curious if we can restricted the wildcard down to something smaller. However, the admissions webhook seems to explicitly require...
```
apiGroup: "*"
resources: "secrets"
verbs: "create"
```

**Review by rmweir — DISMISSED** [review]
Not sure why empty wouldn't work but it seems this is how we add this rule everywhere else. LGTM.

**Comment by nickgerace** [comment]
@rmweir I was confused as well, but I followed the repro steps from the issue, and it worked.

**Comment by nickgerace** [comment]
Had it backwards. The ```user```'s permissions need to be changed instead.

File: pkg/data/management/role_data.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 		addRule().apiGroups("management.cattle.io").resources("kontainerdrivers").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("podsecuritypolicytemplates").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("nodetemplates").verbs("create").
-		addRule().apiGroups("*").resources("secrets").verbs("create").
+		addRule().apiGroups("").resources("secrets").verbs("create").
```
Comment (mrajashree) [inline pkg/data/management/role_data.go:97]: This is changing the existing User role's permissions. If we want to allow restricted admin to create standard user, then [this](https://github.com/rancher/rancher/pull/29575/files#diff-1233e0b4cb129841936b478407cce09e86417b3125fd9e68731243adec2d6003R84) permission should be changed to 
```
addRule().apiGroups("*").resources("secrets").verbs("create").
```

**Review by mrajashree — APPROVED** [review]

**Review by dramich — APPROVED** [review]

**Pull request merged**